src/read/_s3.js

```diff
@@ -105,7 +105,7 @@ module.exports = async function readS3 (params) {
     }
     catch (err) {
       // ETag matches (getObject error code of NotModified), so don't transit the whole file
-      if (err.code === 'NotModified') {
+      if (err.code === 'NotModified' || (err['$metadata'] && err['$metadata'].httpStatusCode === 304)) {
```

Comment:
Looks good! Can make this a bit more terse with optional chaining if you like. Otherwise, merge away!